extension/page-detect.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+window.pageDetect = (() => {
+	const isGistCheck = location.hostname === 'gist.github.com';
+
+	const isGist = () => isGistCheck;
+
+	const isDashboard = () => location.pathname === '/' || /^(\/orgs\/[^\/]+)?\/dashboard/.test(location.pathname);
+
+	const isRepo = () => !isGist() && /^\/[^/]+\/[^/]+/.test(location.pathname);
```

Comment:
Instead of checking against `location.pathname`, we check against a version that has the user/repo stripped away. That way we don't need the first part of the regexes as we already know what we check against is scoped down to the repo.